.github/workflows/android.yml

```diff
@@ -0,0 +1,93 @@
+name: Android CI
+
+on:
+  push:
+    branches:  
+      - master
+      - '!gh-pages'
+    paths:
+      - .github/workflows/**
+      - kotlin/**
+  pull_request:
+    paths:
+      - .github/workflows/**
+      - kotlin/**
+
+jobs:
+  assemble:
+    name: Assemble
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gradle/wrapper-validation-action@v1
+      - name: set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
```

Comment:
This is interesting. I'm not sure what is inside `~/.gradle/caches`, but could there be a race if there're multiple builds in-progress due to to multiple PRs. Could the cache from the `assemble` step from one PR be used in the `check` job in another.

Reply:
I think `Artifacts` is a better action for this: https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts

Reply:
There are no races, caches are stored as tar files, and downloaded to the VM by this action before running your build. But yes, this cache is used so that the assemble job does all the common work needed by the other jobs. To get this to work, you seem to need both the cache configured with the same keys on all jobs, and have the right dependency graph configured via `needs`.

Reply:
I looked at a number of other open source Android projects, as well as the official docs for this action, and this seems to be the standard way of doing it.

Reply:
Gotcha, the cache will be used across different PRs/branches..etc But I guess if `key` hashes the contents of `~/.gradle/caches`, we're all good.

Reply:
We're not hashing the contents of the cache. We're only hashing the contents of the build scripts. The reason for this, and I will document this in the script better, is so that if the build config changes, we start fresh to avoid any bad config polluting the cache. However, gradle does its own caching, so we don't need to clear out the cache if the build configs haven't changed.